File: web/tests/reference/two-pointers/trapping-rain-water.py
def trap(height):
    if not height:
        return 0
    l, r = 0, len(height) - 1
    left_max, right_max, total = height[l], height[r], 0
    while l < r:
        if left_max <= right_max:
            l += 1
            left_max = max(left_max, height[l])
            total += left_max - height[l]
        else:
            r -= 1
            right_max = max(right_max, height[r])
            total += right_max - height[r]
    return total

File: web/tests/reference/two-pointers/trapping-rain-water.js
function trap(height) {
  if (!height.length) return 0;
  let l = 0, r = height.length - 1, lm = height[0], rm = height[height.length - 1], total = 0;
  while (l < r) {
    if (lm <= rm) { l++; lm = Math.max(lm, height[l]); total += lm - height[l]; }
    else { r--; rm = Math.max(rm, height[r]); total += rm - height[r]; }
  }
  return total;
}
